fvm/transaction.go

```diff
@@ -70,65 +70,91 @@ func (i *TransactionInvocator) Process(
 	proc *TransactionProcedure,
 	st *state.State,
 ) error {
-	env, err := newEnvironment(ctx, vm, st)
-	if err != nil {
-		return err
-	}
-	env.setTransaction(proc.Transaction, proc.TxIndex)
 
-	location := common.TransactionLocation(proc.ID[:])
+	var err error
+	var env *hostEnv
+
+	// TODO move me outside
+	numberOfTries := 0
+	maxNumberOfRetries := 2
+	for numberOfTries := 0; numberOfTries < maxNumberOfRetries; numberOfTries++ {
+		env, err = newEnvironment(ctx, vm, st)
+		// env construction error is fatal
+		if err != nil {
+			return err
+		}
+		env.setTransaction(proc.Transaction, proc.TxIndex)
+
+		location := common.TransactionLocation(proc.ID[:])
+
+		err = vm.Runtime.ExecuteTransaction(
+			runtime.Script{
+				Source:    proc.Transaction.Script,
+				Arguments: proc.Transaction.Arguments,
+			},
+			runtime.Context{
+				Interface: env,
+				Location:  location,
+			},
+		)
+
+		// break the loop
+		if !i.requiresRetry(err) {
+			break
+		}
+
+		// reset error part of proc
+		proc.Err = nil
+	}
 
-	err = vm.Runtime.ExecuteTransaction(
-		runtime.Script{
-			Source:    proc.Transaction.Script,
-			Arguments: proc.Transaction.Arguments,
-		},
-		runtime.Context{
-			Interface: env,
-			Location:  location,
-		},
-	)
+	// Note: not sure if this is what we want
+	// panic if we tried several times and still failing
+	if numberOfTries == maxNumberOfRetries {
+		panic(err)
+	}
```

Comment:
did we decide on this ?

Reply:
We want to do this, but can only do so once we have determined that all contracts on mainnet are checking properly, so we can actually detect false negatives (we don't want to crash on true negatives). @janezpodhostnik is working on the check. 

As @Kay-Zee mentioned, let's add this panic in another PR.

Reply:
just commented out that part